hs-bindgen/src/HsBindgen/Lib.hs

```diff
@@ -92,11 +93,11 @@ parseCHeader ::
 parseCHeader tracer args fp =
     WrapCHeader . C.Header <$> C.parseHeaderWith args fp (C.foldDecls tracer)
 
--- | Dump the raw @libclang@ AST
+-- | Show the raw @libclang@ AST
 --
 -- This is primarily for debugging.
-dumpClangAST :: ClangArgs -> FilePath -> IO ()
-dumpClangAST args fp = const () <$> C.parseHeaderWith args fp C.foldDumpAST
+showClangAST :: ClangArgs -> FilePath -> IO (Forest String)
+showClangAST args fp = C.parseHeaderWith args fp C.foldShowAST
```

Comment:
Arguably `parseHeaderWith` and `foldShowAST`  should be part of public API. Maybe not as part of the main module, but these are kind of internal-ish but still generally useful functions (which you need to make a slightly different behavior from what the library provides) that people argue that internal modules should always be exposed.